### PR TITLE
moodle acornfile

### PR DESCRIPTION
**Pull request by slayer321**

### CODE REVIEW

File: Acornfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+services: db: image: "ghcr.io/acorn-io/mariadb:v10.11.5-1"
```
Comment (samkulkarni20) [inline Acornfile]: Use # notation here to always use the latest Acorn service image

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Login to GHCR
+        uses: acorn-io/actions-login@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and Publish with signature
+        run: |
+          TAG=${GITHUB_REF#refs/*/}
+          acorn build --platform linux/amd64 --platform linux/arm64 --push -t ghcr.io/infracloudio/moodle-acorn:${TAG} .
```
Comment (samkulkarni20) [inline .github/workflows/publish.yaml:29]: Did you test out the amd64 & arm64 image creation specified here?

File: Acornfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+containers: {
+	web: {
+		image: "docker.io/bitnami/moodle:4.3"
+		ports: publish: "8080:8080/http"
```
Comment (samkulkarni20) [inline Acornfile]: There are 2 basic things missing here.
1. First of all we should definitely make this service available on secure port with self-signed certs. We shouldn't miss out on this, when it's already done in the reference [docker-compose implementation](https://github.com/bitnami/containers/blob/main/bitnami/moodle/docker-compose.yml#L21) & the dockerfile
2. Secondly, we should have at-least translated the container port 8080 to the default http port i.e. 80. So extending this

**Review by samkulkarni20 — CHANGES_REQUESTED** [review]

**Comment by samkulkarni20** [comment]
Also, where's the readme for this app?

File: Acornfile (hunk 3 above)
Comment (slayer321) [reply Acornfile]: 1. Updated it to use 80 port. Also I observe that using `80:8080`  the URL the acorn platform has https enabled. I think all the certs are by default handle by acorn. When I tried using `443:8443`, after access the URL generated by acorn it gives `Internal Server Error`. So I think just using `80:8080` will work here.

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (slayer321) [reply .github/workflows/publish.yaml:29]: No, will test it once we merge this PR.

File: Acornfile (hunk 1 above)
Comment (slayer321) [reply Acornfile]: I think using specific version will be good as using the `#` will gives us the latest DB, and there might be the case that application version don't support the latest DB version and the application might break.

File: Acornfile [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	serviceArgs: {
+      dbName: args.moodledbname
+  }
+}
+
+containers: {
+	web: {
+		image: "docker.io/bitnami/moodle:4.3"
+		ports: {
+			publish: "80:8080/http"
```
Comment (samkulkarni20) [inline Acornfile:42]: Not yet exposed the https port. That's a must

File: README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Acorn for Moodle
+======
+
+Moodle™ LMS is an open source online Learning Management System widely used at universities, schools, and corporations. It is modular and highly adaptable to any type of online learning.
+
+## Configure the Moodle
+
+Moodle uses mariadb as the database and it is been used as the [services](https://docs.acorn.io/reference/acornfile#services-consuming) . If you are using Advanace Options on Acorn Platform you can provide the `moodledbname`. By default it will be named as `moodledb`.
+By default username and password for moodle is `username: user` `password: bitnami`. You can changes it using `moodle_username` and `moodle_password` field.
```
Comment (samkulkarni20) [inline README.md]: In addition to these, let's also enable configuring [User & Site Configs](https://github.com/bitnami/containers/tree/main/bitnami/moodle#user-and-site-configuration) as well as the [mysql db configs](https://github.com/bitnami/containers/tree/main/bitnami/moodle#create-a-database-for-moodle-using-mysql-client). Of course, not all of the config options in the docker image would make sense for Acornfile. Use your judgement.

**Review by samkulkarni20 — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (slayer321) [reply README.md]: Updated User and Site Config. For DB as we  are using it as  a services so already provided all the field that is mentioned in [Using an Existing Database ](https://github.com/bitnami/containers/tree/main/bitnami/moodle#use-an-existing-database) Section.

File: Acornfile (hunk 4 above)
Comment (slayer321) [reply Acornfile:42]: Even though we have specify `http` here on acorn platform the url that we get has `https` enabled.
I tried giving `https` here if failed with below error 
```
  ✗  ERROR:  schema violation key containers.web.ports: 
               	option 1: [
                   failed to resolve union type to kind [union object] from kind object value ({"publish":"80:8080/https"})
                   failed to resolve union type to kind [union object] from kind object value ({"publish":"80:8080/https"})],
               	option 2: [
               	option 1: [
                   failed to resolve union type to kind [union array] from kind object value ({"publish":"80:8080/https"})
                   failed to resolve union type to kind [union array] from kind object value ({"publish":"80:8080/https"})],
               	option 2: [
               	option 1: [
                   failed to resolve union type to kind [number string] from kind object value ({"publish":"80:8080/https"})
                   failed to resolve union type to kind [number string] from kind object value ({"publish":"80:8080/https"})],
               	option 2: [expected kind number but got object with value ({"publish":"80:8080/https"})],
               	option 3: [expected kind string but got object with value ({"publish":"80:8080/https"})]],

```

Comment (samkulkarni20) [reply Acornfile:42]: Ok

**Review by samkulkarni20 — APPROVED** [review]

**Pull request merged**